sabnzbd/utils/correct_extension.py

```diff
@@ -0,0 +1,342 @@
+#!/usr/bin/python3
+
+""" function to check and find correct extension of a (deobfuscated) file
+"""
+
+
+import puremagic
+import os
+
+# common extension from https://www.computerhope.com/issues/ch001789.htm
+commonextlist = [
+    "3g2",
+    "3gp",
+    "7z",
+    "ai",
+    "aif",
+    "apk",
+    "arj",
+    "asp",
+    "aspx",
+    "avi",
+    "bak",
+    "bat",
+    "bin",
+    "bin",
+    "bmp",
+    "c",
+    "cab",
+    "cda",
+    "cer",
+    "cfg",
+    "cfm",
+    "cgi",
+    "cgi",
+    "cgi",
+    "class",
+    "com",
+    "cpl",
+    "cpp",
+    "cs",
+    "css",
+    "csv",
+    "cur",
+    "dat",
+    "db",
+    "dbf",
+    "deb",
+    "dll",
+    "dmg",
+    "dmp",
+    "doc",
+    "docx",
+    "drv",
+    "email",
+    "eml",
+    "emlx",
+    "exe",
+    "flv",
+    "fnt",
+    "fon",
+    "gadget",
+    "gif",
+    "h",
+    "h264",
+    "htm",
+    "html",
+    "icns",
+    "ico",
+    "ico",
+    "ini",
+    "iso",
+    "jar",
+    "java",
+    "jpeg",
+    "jpg",
+    "js",
+    "jsp",
+    "key",
+    "lnk",
+    "log",
+    "m4v",
+    "mdb",
+    "mid",
+    "midi",
+    "mkv",
+    "mov",
+    "mp3",
+    "mp4",
+    "mpa",
+    "mpeg",
+    "mpg",
+    "msg",
+    "msi",
+    "msi",
+    "odp",
+    "ods",
+    "odt",
+    "oft",
+    "ogg",
+    "ost",
+    "otf",
+    "part",
+    "pdf",
+    "php",
+    "php",
+    "pkg",
+    "pl",
+    "pl",
+    "pl",
+    "png",
+    "pps",
+    "ppt",
+    "pptx",
+    "ps",
+    "psd",
+    "pst",
+    "py",
+    "py",
+    "py",
+    "rar",
+    "rm",
+    "rpm",
+    "rss",
+    "rtf",
+    "sav",
+    "sh",
+    "sql",
+    "svg",
+    "swf",
+    "swift",
+    "sys",
+    "tar",
+    "tar",
+    "gz",
+    "tex",
+    "tif",
+    "tiff",
+    "tmp",
+    "toast",
+    "ttf",
+    "txt",
+    "vb",
+    "vcd",
+    "vcf",
+    "vob",
+    "wav",
+    "wma",
+    "wmv",
+    "wpd",
+    "wpl",
+    "wsf",
+    "xhtml",
+    "xls",
+    "xlsm",
+    "xlsx",
+    "xml",
+    "z",
+    "zip",
+]
+
+downloadextlist = [
```

Comment:
For performance, let's remove all items of `downloadextlist` that are already in `commonextlist`. No need to check anything twice.

Reply:
Please check my implementation to make it unique